### PR TITLE
build: update to django 3.0

### PR DESCRIPTION
**Pull request by macintoshpie**
#### Any background context you want to provide?
SEED is currently on Django 2.2 which is reaching EOL and needs to be updated

#### What's this PR do?
- update django to 3.0 (will update to 3.1 after this merge -- 3.0 is EOL'd, but making the update progressive)
- update django rest framework to 3.12 (latest as of August 2021)
- use xlrd to check for empty workbook instead of pandas (removing pandas from SEED due to slow build time on alpine)
- upgrade other packages for Django 3.x compatibility
- ⚠️ fork jwt-oauth2 and tweak it b/c it breaks on Django 3.x; we need to bring this to the attention of Green Buildings Registry
- remove unused packages
- address some testing warnings (assertEqual, assertDictContainsSubset are deprecated)

I did my best find and address any deprecated or soon to be deprecated (ie 3.1) features by manually testing and running our testsuite [with warnings enabled](https://docs.djangoproject.com/en/3.2/howto/upgrade-version/#resolving-deprecation-warnings):
```
python -Wa manage.py test ...
```
See the output here.

#### How should this be manually tested?
- Spin it up and try miscellaneous SEED features

#### What are the relevant tickets?
#2656

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -1,60 +1,49 @@
 # Django
-django==2.2.24
+django==3.0.14
```
Comment (macintoshpie) [inline requirements/base.txt:2]: party time

File: requirements/base.txt [hunk 2]
```diff
@@ -1,60 +1,49 @@
...
 # Metric/imperial units support
-django-pint==0.4
+django-pint==0.6
 
 # API
-djangorestframework==3.11.2
-# simplejson dependency needs to be installed via github due to a pip segfault with Python 3.6.7
--e git+https://github.com/simplejson/simplejson@v3.16.1#egg=simplejson
-# Django-post-office dependency needs to be installed via github
+djangorestframework==3.12.2
```
Comment (macintoshpie) [inline requirements/base.txt:27]: 🐞

File: requirements/base.txt [hunk 3]
```diff
@@ -69,15 +58,15 @@ geojson==2.5.0
...
 
 # Used by jwt-oauth2
 django-braces==1.14.0
 
-jwt-oauth2>=0.1.1
-django-oauth-toolkit==1.0.0
+# Use forked jwt-oauth2 b/c the published version broke when updating to Django 3.x
+# See: https://docs.djangoproject.com/en/dev/releases/3.0/#removed-private-python-2-compatibility-apis
+# jwt-oauth2>=0.1.1
+-e git+https://github.com/macintoshpie/jwt-oauth2.git@f2db7e16803969f6121129a7b2fa04d09ec5f5e9#egg=jwt-oauth2
```
Comment (macintoshpie) [inline requirements/base.txt:69]: 🍴

Comment (macintoshpie) [reply requirements/base.txt:69]: @nllong can you fork my edits to this repo into the SEED org so yall can maintain it?
https://github.com/macintoshpie/jwt-oauth2

**Review by Ryoken — CHANGES_REQUESTED** [review]
This looks good once that repo is updated!

File: requirements/base.txt [hunk 4]
```diff
@@ -69,15 +58,15 @@ geojson==2.5.0
...
 
 # Used by jwt-oauth2
 django-braces==1.14.0
 
-jwt-oauth2>=0.1.1
-django-oauth-toolkit==1.0.0
+# Use forked jwt-oauth2 b/c the published version broke when updating to Django 3.x
+# See: https://docs.djangoproject.com/en/dev/releases/3.0/#removed-private-python-2-compatibility-apis
+# jwt-oauth2>=0.1.1
+-e git+https://github.com/SEED-platform/jwt-oauth2.git@f2db7e16803969f6121129a7b2fa04d09ec5f5e9#egg=jwt-oauth2
```
Comment (macintoshpie) [inline requirements/base.txt:69]: Now seed owns the jwt fork

**Comment by macintoshpie** [comment]
Merging -- coveralls is failing to accept our push (though all tests passed)

**Pull request merged**